pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -269,6 +269,15 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
         if self.should_rank_save_checkpoint:
             return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
 
+    def remove_checkpoint(self, filepath: str) -> None:
```

Comment:
RIP training type plugin

Reply:
unfortunately i don't see a cleaner way 😞  the training type plugin is covering for all the different backends supported in lightning, and having saving be separate from removing forced the other coupling like needing to expose a property for the callback to act on

at the same time, i think we can better organize the implementations within these plugins because they do fall into clear categories mentioned here: https://docs.google.com/document/d/1xHU7-iQSpp9KJTjI3As2EM0mfNHHr37WZYpDpwLkivA/edit#heading=h.vv5zw27fkkpe